data_pipeline/civicrm_email_report/civicrm_email_report.py

```diff
@@ -0,0 +1,61 @@
+import json
+import datetime
+import logging
+from typing import Iterable
+import requests
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_current_timestamp():
+    return datetime.datetime.utcnow().isoformat()
+
+
+def get_mailing_id_dict(mail_id: int) -> str(dict):
+    return json.dumps({'mailing_id': mail_id})
+
+
+def get_connection_parameters(mail_id: int, api_key, site_key):
+    return {
+        "entity": "Mailing",
+        "action": "stats",
+        "json": get_mailing_id_dict(mail_id),
+        "api_key": api_key,
+        "key": site_key
+    }
+
+
+def get_email_report(
+    url: str,
+    mail_id: int,
+    api_key: str,
+    site_key: str
+) -> dict:
+    response = requests.post(url=url, data=get_connection_parameters(mail_id, api_key, site_key))
+    response.raise_for_status()
+    dict_response = eval(response.text)
```

Comment:
I'd always be careful with using `eval` on anything as it could run any code. Why do we use it here?

Reply:
I found another way to doing it. What about this one?
```python
import ast
ast.literal_eval("{'key1' : 'value1', 'key2' : 'value2'}")
```

```
literal_eval(node_or_string)
    Safely evaluate an expression node or a string containing a Python
    expression.  The string or node provided may only consist of the following
    Python literal structures: strings, numbers, tuples, lists, dicts, booleans,
    and None.
```

Reply:
used json.loads()